pkg/protocompat/time_test.go

```diff
@@ -115,6 +115,27 @@ func TestConvertTimeToTimestampOrNil(t *testing.T) {
 	assert.Nil(t, protoTSInvalid)
 }
 
+func TestConvertTimestampToGraphqlTimeOrError(t *testing.T) {
+	gqlTime, err := ConvertTimestampToGraphqlTimeOrError(nil)
+	assert.NoError(t, err)
+	assert.Nil(t, gqlTime)
+
+	protoTime := &types.Timestamp{
+		Seconds: 2345678901,
+		Nanos:   234567891,
+	}
+	gqlTime, err = ConvertTimestampToGraphqlTimeOrError(protoTime)
+	assert.NoError(t, err)
+	assert.Equal(t, int64(2345678901234567891), gqlTime.UnixNano())
+
+	invalidProtoTime := &types.Timestamp{
+		Seconds: 253402300801,
```

Comment:
We should probably use constants with appropriate comments to represent valid `Seconds` and `Nanos` values within `Timestamp` and `Duration` objects.

Values for `Timestamp` are (https://github.com/gogo/protobuf/blob/f67b8970b736e53dbd7d0a27146c8f1ac52f74e5/types/timestamp.go#L43-L48) :
minValidSeconds: -62135596800 // time.Date(1,1,1,0,0,0,0, time.UTC).Unix()
maxValidSeconds: 253402300800 // time.Date(10000,1,1,0,0,0,0,time.UTC).Unix()

The given values for Duration seem to not match what provides the actual outcome of the validation function.
https://github.com/gogo/protobuf/blob/f67b8970b736e53dbd7d0a27146c8f1ac52f74e5/types/duration.go#L44-L47
The expected value would be 10000 years. With the overflow check taken into account, the values I found as being the boundaries are:
maxDurationSeconds = 9223372036
minDurationSeconds = -9223372036

Reply:
I have changed tests. I don't want to use edge values.

One framework uses `253402300800` (gogo proto).
Another uses `253402300799`(google proto).

I'm using "common" sense validation cases. And I have added comments with explanations. (no constants)